test/unit/transforms/deployments.test.ts

```diff
@@ -0,0 +1,59 @@
+import _ from "lodash";
+import { mapEnvironment } from "../../../src/transforms/deployment";
+
+describe("deployment environment mapping", () => {
+	const exampleEnvironmentNames = [
+		"development",
+		"dev",
+		"trunk",
+		"dev",
+		"testing",
+		"test",
+		"tests",
+		"tst",
+		"integration",
+		"integ",
+		"intg",
+		"int",
+		"acceptance",
+		"accept",
+		"acpt",
+		"qa",
+		"qc",
+		"control",
+		"quality",
+		"staging",
+		"stage",
+		"stg",
+		"preprod",
+		"model",
+		"internal",
+		"production",
+		"prod",
+		"live",
+		"some-random-name",
+		"internet"
+	]
+	// Variations of the example names, e.g. "prod-east" is considered a variation of "prod"
+	const exampleEnvironmentNameVariations = _.flatMap(exampleEnvironmentNames, environmentName => [
+		environmentName,
+		// Different case
+		environmentName.toUpperCase(),
+		// Add some prefixes & postfixes
+		`${environmentName}-east`,
+		`${environmentName}-eu-west`,
+		`east-${environmentName}`,
+		`jane-${environmentName}-local`,
+		// Check some other common separators
+		`${environmentName}:east`,
+		`${environmentName}_east`,
+		`${environmentName}/east`,
+		`${environmentName} east`,
+	]);
+
+	it("should transform commonly used environment names to the expected environment types", () => {
+		const environmentTypesToEnvironmentNames = _.groupBy(exampleEnvironmentNameVariations, mapEnvironment);
+
+		expect(environmentTypesToEnvironmentNames).toMatchSnapshot();
```

Comment:
Can't say I'm a fan of having a test just based on the snapshot as it can easily be updated by mistake and there's nothing else to catch it.  It will also fail if any new environments are added or regex might change the capture, but not specifically going through what are the 'rules' of the algorithm.

I also don't see any failure tests, only successful ones.

Reply:
Yeah that's fair, snapshot tests do have their drawbacks. I'll rework the tests 👍 

> I also don't see any failure tests, only successful ones.

I was thinking the [unmapped environments](https://github.com/atlassian/github-for-jira/pull/691/files#diff-bd38d697968a2846e77fea1a253983dd9944eb96b2633953f6af5b1e03a319f7R293) could be considered failures in this context. Perhaps that's not what you mean though?

Reply:
I guess, but those failures are very simplistic.  they don't have 'prod' or any other environments in their naming convention at all.  I'd probably try things like `postage` or have some names with some non-latin characters, etc.  They will return as unmapped, but the more environment names we can throw at it, the more confidence we can have in the regex :)

Reply:
Yep good idea, I've added a non-latin example. I think I'm missing the significance of `postage` though?

Reply:
`postage` has `stage` in it :P